sleap/io/cameras.py

```diff
@@ -853,13 +864,62 @@ def update_points(
             exclude_complete: If True, then do not update points that are marked as
                 complete. Default is True.
         """
+        # Ensure we are working with a float array
+        points_3d = points_3d.astype(float)
+
+        # Check if points are 3D
+        is_3d = points_3d.shape[-1] == 3
+        if not is_3d:
+            raise ValueError("Expected 3D points with shape (M, T, N, 3).")
+        
+        # Check that the correct shape was passed in
+        n_views, n_instances, n_nodes, n_coords = points_3d.shape
+        assert n_views == len(
+            self.cams_to_include
+        ), f"Expected {len(self.cams_to_include)} views, got {n_views}."
+        assert n_instances == len(
+            instance_groups
+        ), f"Expected {len(instance_groups)} instances, got {n_instances}."
+        assert n_coords == 3, f"Expected 3 coordinates, got {n_coords}."
+
+        # Reproject 3D points into 2D points for each camera view
+        pts_reprojected = reproject(
+            points_3d,
+            calib=self.session.camera_cluster,
+            excluded_views=self.excluded_views,
+        )  # M=include x F=1 x T x N x 2
+
+        # Squeeze back to the original shape
+        points_reprojected = np.squeeze(pts_reprojected, axis=1)  # M=include x TxNx2
+
+        # Get projection bounds (based on video height/width)
+        bounds = self.session.projection_bounds
+        bounds_expanded_x = bounds[:, None, None, 0]
+        bounds_expanded_y = bounds[:, None, None, 1]
+
+        # Create masks for out-of-bounds x and y coordinates
+        out_of_bounds_x = (points_reprojected[..., 0] < 0) | (points_reprojected[..., 0] > bounds_expanded_x)
+        out_of_bounds_y = (points_reprojected[..., 1] < 0) | (points_reprojected[..., 1] > bounds_expanded_y)
+
+        # Replace out-of-bounds x and y coordinates with nan
+        points_reprojected[out_of_bounds_x, 0] = np.nan
+        points_reprojected[out_of_bounds_y, 1] = np.nan
+
+        # Update points for each `InstanceGroup`
+        for ig_idx, instance_group in enumerate(instance_groups):
+            # Ensure that `InstanceGroup`s is in this `FrameGroup`
+            self._raise_if_instance_group_not_in_frame_group(
+                instance_group=instance_group
+            )
+            # Update points for the instance group
+            instance_group.points = points_reprojected[ig_idx]
```

Comment:
We will just be passing in points for a single `InstanceGroup`, so we won't be needing this here (but, this will still be similar to how we call `InstanceGroup.update_points` from `FrameGroup.upsert_points`).
```suggestion
```